morbusgame/gamemode/server/sv_maps.lua

```diff
@@ -77,149 +77,96 @@ util.AddNetworkString("smv_winner")
 
 
 SMV.VoteTime = 30
-SMV.OptionCount = 4
 SMV.Voting = false
 SMV.Votes = {}
 SMV.TVotes = {}
 
 SMV.MapVoteList = {}
 
 function SMV.CreateMapList()
-	SMV.ExcludedMaps = {}
 	
-	--read prevous maps and convert to table format
-	local lastMaps = util.JSONToTable( file.Read( "lastmaps.txt") )
+	--build list of maps to exclude based on size and get exclude string 
+	SMV.ExcludedMaps, excludeString = SMV.CreateExcludeList()
 	
-	--insert the current map into the table
-	local currentMap = game.GetMap()
-	table.insert(lastMaps, 1, currentMap)
+	--clear table from previous round (safety feature)
+	table.Empty(SMV.MapVoteList)
 	
-	--exclude maps based on player count	
-	local playerCount = GetValidCount()	
+	--Build list of maps to vote (check for excludes and the map is installed)
+	for k,v in pairs(SMV.Maps) do
+		if !table.KeyFromValue(SMV.ExcludedMaps, v) then
+			if file.Exists("maps/"..v..".bsp","GAME") then
+				table.insert(SMV.MapVoteList, v)
+			end		
+		end	
+	end
 	
+	--send map vote when done
+	net.Start("smv_start")
+	net.WriteTable(SMV.MapVoteList)
+	net.Broadcast()
+	
+	--let players know about our map rotions. Do it at the bottom incase if there's an error.
+	SMV.SendAll("[xG] There are "..table.Count(SMV.Maps).." total maps in rotations!")
+	SMV.SendAll("[xG] Player count thresholds have excluded "..excludeString.." maps!")	
+end
+
+
+--exclude maps based on player count	
+function SMV.CreateExcludeList()
+	SMV.ExcludedMaps = {}
+	local playerCount = GetValidCount()	
 	local excludeString = ""
 	
-	if playerCount < 18 then
+	if playerCount < 20 then
 		table.insert(SMV.ExcludedMaps, "mor_horizon_v11_re")
 		
 		if playerCount < 12 then	
 			table.insert(SMV.ExcludedMaps, "mor_installation_gt1_re")
 			table.insert(SMV.ExcludedMaps, "mor_outpostnorth32_a5")
 			table.insert(SMV.ExcludedMaps, "mor_auriga_v4_re")
-			table.insert(SMV.ExcludedMaps, "mor_ptmc_v22")				
+			table.insert(SMV.ExcludedMaps, "mor_ptmc_v22")
+			table.insert(SMV.ExcludedMaps, "mor_breach_cv21")
```

Comment:
Breach_cv21 should only be excluded at 12 players, if previous maps exclusion is disabled. There will not be enough options in the map vote otherwise. Add if statement here to check feature status before excluding.

Reply:
Agree